### PR TITLE
Add parallel action execution for `ToolkitTask`

### PR DESCRIPTION
**Pull request by vasinov**
This PR introduces a new way for LLMs to execute actions. LLMs accessed via `ToolkitTask` can now execute multiple actions in parallel. Griptape will execute all provided actions with a `futures_executor`. The following is a list of major changes:

* Changes to the `ToolkitTask` system prompt explianing how to run actions in parallel.
* `ActionSubtask` was renamed to `ActionsSubtask`.
* Split action execution logic into `execute_actions` and `execute_action` in `ActionsSubtask`.
* Added `BaseActionSubtaskEvent.subtask_actions`.
* Removed `subtask_action_name`, `subtask_action_path`, and `subtask_action_input` in `BaseActionSubtaskEvent`.
* Improved system prompt in`ToolTask` to support more use cases.

This change was tested on gpt-4 and Claude2 with temperatures ranging from 0.1 to 0.9.

### CODE REVIEW

**Comment by shhlife** [comment]
Let’s make sure we update Grippy’s logic before we merge this to main - since it’ll change how it handles subtasks

File: griptape/templates/tasks/toolkit_task/system.j2
Comment (collindutter) [inline griptape/templates/tasks/toolkit_task/system.j2:4]: Is this redundant? Including the schemas twice feels like it could confuse the LLM.

File: griptape/tasks/actions_subtask.py
Comment (collindutter) [inline griptape/tasks/actions_subtask.py:28]: Should this be moved inside of `ActionsSubtask` similar to how we have `PromptStack.Input`?

File: griptape/events/base_action_subtask_event.py
Comment (collindutter) [inline griptape/events/base_action_subtask_event.py:12]: Should this be renamed to `BaseActionsSubtaskEvent`?

File: griptape/mixins/action_subtask_origin_mixin.py
Comment (collindutter) [inline griptape/mixins/action_subtask_origin_mixin.py:1]: Should this be renamed to `ActionsSubtaskOriginMixin`?

**Review by collindutter — COMMENTED** [review]
Nice work! Do we want to take this opportunity to standardize Action/Tool naming? I think in the past we had talked about renaming Action -> Tool.

**Review by collindutter — DISMISSED** [review]
Approved once @shhlife verifies his tests work.

**Comment by collindutter** [comment]
Oh, just noticed failing type-check. Can you please resolve that?

**Review by collindutter — APPROVED** [review]

**Pull request merged**
